niceman/distributions/debian.py

```diff
@@ -89,26 +89,26 @@ class APTSource(SpecObject):
 _register_with_representer(APTSource)
 
 
-@attr.s(slots=True)
+@attr.s(slots=True, frozen=True, cmp=False, hash=True)
 class DEBPackage(Package):
     """Debian package information"""
-    name = attr.ib()
+    name = attr.ib(hash=True)
     # Optional
-    source_name = attr.ib(default=None)
+    source_name = attr.ib(default=None, hash=False)
     upstream_name = attr.ib(default=None)
 
-    version = attr.ib(default=None)
-    architecture = attr.ib(default=None)
-    size = attr.ib(default=None)
-    md5 = attr.ib(default=None)
-    sha1 = attr.ib(default=None)
-    sha256 = attr.ib(default=None)
-    source_version = attr.ib(default=None)
-    versions = attr.ib(default=None)
-    install_date = attr.ib(default=None)
+    version = attr.ib(default=None, hash=True)
+    architecture = attr.ib(default=None, hash=False)
```

Comment:
there could be multiple instances of the package with the same name and version but different architecture
```shell
$> dpkg -l zlib1g:*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                 Version         Architecture    Description
+++-====================-===============-===============-==============================================
ii  zlib1g:amd64         1:1.2.8.dfsg-5  amd64           compression library - runtime
ii  zlib1g:i386          1:1.2.8.dfsg-5  i386            compression library - runtime
```
so should participate in the hashing